### PR TITLE
chore: Forge v6 blog post

### PR DESCRIPTION
**Pull request by georgexu99**
initial PR to get things going ahead of forge v6 release.

## Questions/ areas of note

- I refer to forge v6 as: Electron Forge v6, Forge, Forge v6 let me know if this should be standardized
- breaking changes needs to be fleshed out

### CODE REVIEW

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:27]: Forge v5 is pretty dead already compared to Forge v6, since we've been pushing the betas for years now: https://npmtrends.com/@electron-forge/core-vs-electron-forge

The number of people switching from v5 would drastically be outnumbered by:
* Developers using v6 beta and wanting to learn about what "stable" means for the project
* Developers using Builder and wanting to migrate over
* Developers using their own packager/osx-sign/etc. pipeline and looking to migrate over

IMO this blog post should be rewritten to those three crowds rather than for a v5 -> v6 migration.

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:68]: What is this changelog comparing to? It's not very relevant if it's from v5 since v6 was a complete rewrite

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:2]:
```suggestion
title: Electron Forge v6.0.0
```

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:23]: > Now that we are done making changes

Is this the answer to "What Does Switching to Stable Mean"? Doesn't seem right to me.

**Review by erickzhao — CHANGES_REQUESTED** [review]

File: blog/forge-v6-release.md
Comment (georgexu99) [reply blog/forge-v6-release.md:27]: 👀 really good point, we haven't addressed points 2 and 3 much.

File: blog/forge-v6-release.md
Comment (georgexu99) [reply blog/forge-v6-release.md:23]: "now that we are done making _major breaking_ changes"?

File: blog/forge-v6-release.md
Comment (georgexu99) [reply blog/forge-v6-release.md:68]: Changelog indeed refers to v5 -> v6. Should we have a breaking changes / changelog section at all then?

Comment (erickzhao) [reply blog/forge-v6-release.md:68]: I would add the breaking changes from the final beta IMO.

File: blog/forge-v6-release.md
Comment (georgexu99) [reply blog/forge-v6-release.md:27]: Adding a benefits of using electron forge section- is community engagement and beta testing from larger companies ie. slack something we want to push?

 think forge is ~1k weekly while core/cli are ~30. Should I axe the migration section

Comment (erickzhao) [reply blog/forge-v6-release.md:27]: > Adding a benefits of using electron forge section- is community engagement and beta testing from larger companies ie. slack something we want to push?

I think one of the main points is that Forge will be kept up to date with the newest Electron tooling updates (e.g. universal builds, ASAR integrity, etc.). If you're coming from Builder, you don't have to wait for their maintainers to implement it. If you're coming from your own tooling pipeline, you don't have to figure out how to wire in the features yourselves.

Comment (erickzhao) [reply blog/forge-v6-release.md:27]: > think forge is ~1k weekly while core/cli are ~30. Should I axe the migration section

I think Forge v5 migration isn't that useful. Migrating from an existing project is probably something that you want to include. It should be better documented as well in our Forge docs, since all we have is "run the import script and hope it works".

**Comment by VerteDinde** [comment]
We should probably add a "Migrating from v5 to v6" section, just to outline the major changes people will need to look out for.

**Comment by georgexu99** [comment]
> We should probably add a "Migrating from v5 to v6" section, just to outline the major changes people will need to look out for.

I removed this section as per Erick's first comment- given that 1/30 of our users are using v5, and that the migration guide basically says "run the script, resolve any remaining conflicts in the config" it may not be valuable in the blog

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:61]: general formatting thing: we usually do Title Case for the blog title but Sentence case for headings
```suggestion
## A call to action:
```

File: blog/forge-v6-release.md
Comment (VerteDinde) [inline blog/forge-v6-release.md:51]: Let's do a pass of this by Wednesday EOD to make sure we got all breaking changes:
* Listed here
* Updated in docs

File: blog/forge-v6-release.md
Comment (VerteDinde) [inline blog/forge-v6-release.md:156]:
```suggestion
Tell us what you need! The Electron Forge Team is always looking to build the project to better suit its users. 

You can help us improve Electron Forge by submitting feature requests, posting [issues](https://github.com/electron/forge/issues), or just letting us know your feedback! You can also join us in the [official Electron Discord server](https://discord.com/invite/electronjs), where there is a dedicated channel for Electron Forge.
```

Comment (georgexu99) [reply blog/forge-v6-release.md:156]: Thanks! I remember pushing this discord link up on friday but i guess somewhere it got lost :P

File: blog/forge-v6-release.md
Comment (MarshallOfSound) [inline blog/forge-v6-release.md:36]:
```suggestion
2. *Forge's multi-package architecture makes it easy to understand and extend.* Forge was built with the intention of supporting custom plugins, makers and publishers. Since Forge is made up of many smaller packages with clear responsibilities, it is easier to follow code flow. In addition, Forge's extensible API design means that you can write your own additional build logic separate from the provided configuration options for advanced use cases. For more details on extensibility, see the [Extending Electron Forge] section of the docs, or visit the ["Why Electron Forge"](https://www.electronforge.io/core-concepts/why-electron-forge) section of our docs to learn more about the benefits that Forge offers.
```

File: blog/forge-v6-release.md
Comment (MarshallOfSound) [inline blog/forge-v6-release.md:54]: Forge v5 had a CLI, it's only the core API that's new.  I would remove this bit

File: blog/forge-v6-release.md
Comment (MarshallOfSound) [inline blog/forge-v6-release.md:111]:
```suggestion
#### Prefer forge.config.js over package.json config ([#2991](https://github.com/electron-userland/electron-forge/commit/777197e5))
```

File: blog/forge-v6-release.md
Comment (MarshallOfSound) [inline blog/forge-v6-release.md:121]:
```suggestion
#### Upgraded Required Node.js to 14 LTS ([#2921](https://github.com/electron/forge/pull/2921))
```

File: blog/forge-v6-release.md
Comment (MarshallOfSound) [inline blog/forge-v6-release.md:131]:
```suggestion
For consistency with the `packagerConfig` option for `electron-packager`, the field to configure `@electron/rebuild` has now been shortened to `rebuildConfig`.
```

File: blog/forge-v6-release.md
Comment (MarshallOfSound) [inline blog/forge-v6-release.md:151]: Should we point people at `template-webpack-typescript` instead?

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:28]: I would add an Installation section before this maybe?

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:56]: Since we're using the same `import` command for Forge v5 and non-Forge apps we should make this part of the aforementioned Installation section.

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:54]: This should be a small detail in the Installation section rather than its own section.

If we want to talk about Core commands and CLI, we could make that part of a "What is Forge" overview section.

**Review by erickzhao — CHANGES_REQUESTED** [review]

File: blog/forge-v6-release.md
Comment (VerteDinde) [inline blog/forge-v6-release.md:43]: Super nit:
```suggestion
This section assumes you have a pre-existing project. If you are looking to get started with a new Forge project, head on to the [Getting Started] section in the docs. 
```

File: blog/forge-v6-release.md
Comment (VerteDinde) [inline blog/forge-v6-release.md:26]:
```suggestion
Visit the ["Why Electron Forge"](https://www.electronforge.io/core-concepts/why-electron-forge) section of our docs to learn more about Forge and the benefits that it offers.
```

**Review by VerteDinde — COMMENTED** [review]
Tentative ✅  - going to hold off approving until we're ready to merge (right after stable)

File: blog/forge-v6-release.md
Comment (BlackHole1) [inline blog/forge-v6-release.md:80]:
```suggestion
2. *Forge's multi-package architecture makes it easy to understand and extend.* Forge was built with the intention of supporting custom plugins, makers and publishers. Since Forge is made up of many smaller packages with clear responsibilities, it is easier to follow code flow. In addition, Forge's extensible API design means that you can write your own additional build logic separate from the provided configuration options for advanced use cases. For more details on extensibility, see the [Extending Electron Forge] section of the docs.
```

**Review by BlackHole1 — CHANGES_REQUESTED** [review]

File: blog/forge-v6-release.md
Comment (erickzhao) [reply blog/forge-v6-release.md:80]: See https://github.com/electron/electronjs.org-new/pull/278#discussion_r1009869856 😄

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:47]: We say that migrating from v5 is "the same process" as importing an existing app to Forge, but this is a bit weirdly-phrased because we don't explain what that "same" process is.

Last sentence isn't too useful since we have the code snippet already.

```suggestion
When you use the `import` command, Electron Forge will add a few core dependencies and create a new `forge.config.js` configuration. If you have any existing build tooling (e.g. Electron Packager, Electron Builder, Forge v5), it will try to migrate as many settings as possible. Some of your existing configuration may need to be migrated manually.
```

I would also move this command below the Tabs.

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:70]:
```suggestion
Manual migration details can be found in the Forge [import documentation]. If you need help, please stop by the Electron [Discord](https://discord.gg/f4cH9BzaDw) server.
```

less passive tone

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:74]: I'd be more general in the intro sentence

```suggestion
If you already have tooling for packaging and publishing your Electron app, the benefits associated with adopting Electron Forge can still outweigh the initial setup cost.
```

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:76]:
```suggestion
We believe there are two main benefits to using Forge:
```

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:78]:
```suggestion
1. *Forge receives new features for application building as soon as they are supported in Electron*. This means that alongside the implicit features and plugins that Forge provides, users won't need to wire in new tooling support themselves, or wait for that support to be eventually implemented by other packages before upgrading. For recent examples, see [`@electron/universal`](https://github.com/electron/universal) and [ASAR Integrity checking](https://www.electronjs.org/docs/latest/tutorial/asar-integrity).
```

Getting straight to the point

**Review by erickzhao — CHANGES_REQUESTED** [review]

File: blog/forge-v6-release.md
Comment (georgexu99) [reply blog/forge-v6-release.md:74]: Gonna keep the word `switching` cost, I think it first the definition of what the user is doing a bit more (nit)
 feel free to push back if you feel strongly abt it though

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:65]: > an ultra-minimal Electron app boilerplate inside

I don't think that's accurate with the `--template=webpack` flag enabled

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:69]: > , which we have slipped into the initialization commands above.

would prefer explaining it above rather than saying that here

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:75]:
```suggestion
This plugin integrates webpack with Electron Forge in a few ways:
- Enhanced local development flow with `webpack-dev-server`, including support for Hot Module Replacement in the renderer.
- Handles build logic for webpack bundles before the Package step.
- Support for Native Node modules in the webpack bundling process
```

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:114]:
```suggestion
## Breaking changes
```

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:116]: Maybe specify the exact beta range you're pulling from

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:190]:
```suggestion
## Feedback wanted!
```

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:173]: maybe mention this is for the osxSign part of the config

**Review by erickzhao — CHANGES_REQUESTED** [review]

File: blog/forge-v6-release.md
Comment (georgexu99) [reply blog/forge-v6-release.md:75]: > This plugin integrates webpack with Electron Forge in a few ways, including:
- Enhancing the local development flow with `webpack-dev-server`, including support for Hot Module Replacement in the renderer,
- Handling build logic for webpack bundles before the Package step, and,
- Adding support for Native Node modules in the webpack bundling process.

modifying the styling for sentence-style bullet lists

File: blog/forge-v6-release.md
Comment (georgexu99) [reply blog/forge-v6-release.md:190]: I think we should stick with `a call to action`, or modify this to `Submit your Feedback to us!` 

- employing a push-style call to action phrase is better to incite desired action from readers

File: blog/forge-v6-release.md
Comment (georgexu99) [reply blog/forge-v6-release.md:173]: can you elaborate? I'm not sure I follow

File: blog/forge-v6-release.md
Comment (erickzhao) [reply blog/forge-v6-release.md:189]: I wasn't sure if "A call to action" was a placeholder for "add a CTA here" LOL

File: blog/forge-v6-release.md
Comment (erickzhao) [reply blog/forge-v6-release.md:172]: This property is pretty deep in the Forge config.

```js
module.exports = {
 packagerConfig: {
   osxSign: {
     identityValidation: 'foo'
   }
 }
}
```

It'd be good to make that clearer

Comment (erickzhao) [reply blog/forge-v6-release.md:172]: On second thought, I actually think there are more breaking changes in the `@electron/osx-sign` migration than just this one property, so I think this section is misleading.

We can flesh this out tomorrow but the tool's default code signing entitlements are different (and should work out of the box without additional configuration) and the way we set entitlements is different as well I think.

A hacky 1:1 migration looks something like this https://github.com/electron/osx-sign/blob/main/MIGRATION.md#direct-migration

File: blog/forge-v6-release.md
Comment (erickzhao) [reply blog/forge-v6-release.md]: > sentence-style bullet lists

I don't use these often but various style guides I see online punctuate this differently: no capital letters since it's the same sentence, semicolons between bullet points, and no comma on the last `and`.

> This plugin integrates webpack with Electron Forge in a few ways, including:
> 
> * enhancing the local development flow with webpack-dev-server, including support for Hot Module Replacement in the renderer;
> * handling build logic for webpack bundles before the Package step; and
> * adding support for Native Node modules in the webpack bundling process.

Comment (georgexu99) [reply blog/forge-v6-release.md]: hmm i guess I learned a diff style guide, but happy with this one, generally a fan of sentence-style bullet list

Comment (georgexu99) [reply blog/forge-v6-release.md]: on further review yeah good catch- the style guide I learned was with semicolons and lowercase (though i dont think the comma at the end of `and,` really matteres

happy with using this formatting since I generally prefer sentence-style bullets

File: blog/forge-v6-release.md
Comment (georgexu99) [reply blog/forge-v6-release.md:189]: LOL v understandable- I'm gonna change this to `Submit your feedback!` so readers don't think the same way

File: blog/forge-v6-release.md
Comment (georgexu99) [reply blog/forge-v6-release.md]: Good catch- I've moved this breaking change to be a subheading within the `electron-packager@17` changes so that it's more logically cohesive with the `osx-sign` changes as a preliminary step. I've also added a title in the diff for `identityValidation` so that it specifies where in the config it lies.

We should do another review to flesh this section out and make sure we catch all breaking changes tomorrow morning though.

File: blog/forge-v6-release.md
Comment (erickzhao) [inline blog/forge-v6-release.md:3]:
```suggestion
date: 2022-11-03T00:00:00.000Z
```

**Review by VerteDinde — APPROVED** [review]

File: blog/forge-v6-release.md
Comment (VerteDinde) [inline blog/forge-v6-release.md:57]:
```suggestion
  npm init electron-app@latest my-app --template=webpack
```

**Review by erickzhao — APPROVED** [review]

File: blog/forge-v6-release.md
Comment (erickzhao) [reply blog/forge-v6-release.md]: [ff67c09](https://github.com/electron/electronjs.org-new/pull/278/commits/ff67c0909120f29a46fbc9cf9aa0bfe69ed577cd)

**Comment by georgexu99** [comment]
WOO 🍾

**Pull request merged**
